src/workerd/jsg/resource.h

```diff
@@ -191,6 +214,308 @@ template <typename TypeWrapper,
+    if constexpr (trackCallCount) {
+      jsg::fast_api::callCounter.slow++;
+    }
+    MethodCallback<TypeWrapper, methodName, isContext, T, Ret (U::*)(Args...), method,
+        kj::_::Indexes<indexes...>>::callback(args);
+  }
+
+  template <bool trackCallCount = false>
+  static Ret fastCallback(
+      v8::Local<v8::Object> receiver, Args... fastArgs, v8::FastApiCallbackOptions& options) {
+
+    if constexpr (trackCallCount) {
+      jsg::fast_api::callCounter.fast++;
+    }
+    auto isolate = options.isolate;
+    v8::HandleScope handleScope(isolate);
+    auto context = isolate->GetCurrentContext();
+    auto& self = extractInternalPointer<T, isContext>(context, receiver);
+
+    return executeFastMethod<Ret>(isolate, [&]() -> Ret {
+      if constexpr (std::is_same_v<Ret, void>) {
+        (self.*method)(fastArgs...);
+        return;
+      } else {
+        return (self.*method)(fastArgs...);
+      }
+    });
+  }
+
+  template <bool trackCallCount = false>
+  static v8::CFunction makeCFunction() {
+    static_assert(IsFastApiCompatible<Ret, Args...>::value);
+    return v8::CFunction::Make(fastCallback<trackCallCount>);
+  }
+};
+
+// Specialization for const instance methods compatible with Fast API
+template <typename TypeWrapper,
+    const char* methodName,
+    bool isContext,
+    typename T,
+    typename U,
+    typename Ret,
+    typename... Args,
+    Ret (U::*method)(Args...) const,
+    size_t... indexes>
+struct FastMethodCallback<TypeWrapper,
+    methodName,
+    isContext,
+    T,
+    Ret (U::*)(Args...) const,
+    method,
+    kj::_::Indexes<indexes...>> {
+
+  template <bool trackCallCount = false>
+  static void slowCallback(const v8::FunctionCallbackInfo<v8::Value>& args) {
+    if constexpr (trackCallCount) {
+      jsg::fast_api::callCounter.slow++;
+    }
+    MethodCallback<TypeWrapper, methodName, isContext, T, Ret (U::*)(Args...) const, method,
+        kj::_::Indexes<indexes...>>::callback(args);
+  }
+
+  template <bool trackCallCount = false>
+  static Ret fastCallback(
+      v8::Local<v8::Object> receiver, Args... fastArgs, v8::FastApiCallbackOptions& options) {
+
+    if constexpr (trackCallCount) {
+      jsg::fast_api::callCounter.fast++;
+    }
+    auto isolate = options.isolate;
+    v8::HandleScope handleScope(isolate);
+    auto context = isolate->GetCurrentContext();
+    auto& self = extractInternalPointer<T, isContext>(context, receiver);
+
+    return executeFastMethod<Ret>(isolate, [&]() -> Ret {
+      if constexpr (std::is_same_v<Ret, void>) {
+        (self.*method)(fastArgs...);
+        return;
+      } else {
+        return (self.*method)(fastArgs...);
+      }
+    });
+  }
+
+  template <bool trackCallCount = false>
+  static v8::CFunction makeCFunction() {
+    static_assert(IsFastApiCompatible<Ret, Args...>::value);
+    return v8::CFunction::Make(fastCallback<trackCallCount>);
+  }
+};
+
+// Specialization for const instance methods
```

Comment:
This doesn't seem related to fast API? Are you just adding support for const methods because they weren't supported before?

Do we actually need this? Can it be in a separate PR, to reduce the complexity of this PR?